integration/test/test_cpu_characterization.py

```diff
@@ -0,0 +1,207 @@
+
+        # Assign all cores to resource monitoring association ID 0. This
+        # allows for monitoring the resource usage of all cores.
+        geopm_test_launcher.geopmwrite("MSR::PQR_ASSOC:RMID board 0 {}".format(0))
+        # Assign the resource monitoring ID for QM Events to match the per
+        # core resource association ID above (0)
+        geopm_test_launcher.geopmwrite("MSR::QM_EVTSEL:RMID board 0 {}".format(0))
+        # Select monitoring event ID 0x2 - Total Memory Bandwidth Monitoring.
+        # This is used to determine the Xeon Uncore utilization.
+        geopm_test_launcher.geopmwrite("MSR::QM_EVTSEL:EVENT_ID board 0 {}".format(2))
+
+        # Grabbing system frequency parameters for experiment frequency bounds
+        cls._cpu_base_freq = geopm_test_launcher.geopmread("CPU_FREQUENCY_STICKER board 0")
+
+        #TODO: dropping max freq for max many core freq?
+        #cls._cpu_max_freq = geopm_test_launcher.geopmread("CPU_FREQUENCY_MAX_AVAIL board 0")
+        cls._cpu_max_freq = geopm_test_launcher.geopmread("MSR::TURBO_RATIO_LIMIT:MAX_RATIO_LIMIT_7 board 0")
+        cls._uncore_max_freq = geopm_test_launcher.geopmread("CPU_UNCORE_FREQUENCY_MAX_CONTROL board 0")
+
+        cls._cpu_min_freq = geopm_test_launcher.geopmread("CPU_FREQUENCY_MIN_AVAIL board 0")
+        cls._uncore_min_freq = geopm_test_launcher.geopmread("CPU_UNCORE_FREQUENCY_MIN_CONTROL board 0")
+
+        node_count = 1
+
+        mach = machine.init_output_dir('.')
+
+        def launch_helper(experiment_type, experiment_args, app_conf, experiment_cli_args):
+            output_dir = experiment_args.output_dir
+            if output_dir.exists() and output_dir.is_dir():
+                shutil.rmtree(output_dir)
+
+            experiment_type.launch(app_conf=app_conf, args=experiment_args,
+                                   experiment_cli_args=experiment_cli_args)
+
+        #####################
+        # Setup Common Args #
+        #####################
+        geopm_test_launcher.geopmwrite("CPU_FREQUENCY_MAX_CONTROL board 0 {}".format(cls._cpu_max_freq))
+        geopm_test_launcher.geopmwrite("CPU_UNCORE_FREQUENCY_MIN_CONTROL board 0 {}".format(cls._uncore_min_freq))
+        geopm_test_launcher.geopmwrite("CPU_UNCORE_FREQUENCY_MAX_CONTROL board 0 {}".format(cls._uncore_max_freq))
+
+        experiment_args = SimpleNamespace(
+            node_count=node_count,
+            trial_count=3,
+            cool_off_time=3,
+            enable_traces=False,
+            enable_profile_traces=False,
+            verbose=False,
+        )
+
+        # Arithmetic Intensity Benchmark
+        # Only benchmarks 1 & 16 are used to reduce characterization time
+        aib_app_conf = arithmetic_intensity.ArithmeticIntensityAppConf(
+            ['--slowdown=1',
+             '--base-internal-iterations=10',
+             '--iterations=5',
+             f'--floats={1<<21}', #quick characterization, use 26 for slower & more accurate
```

Comment:
This problem size is determined empirically and is not guaranteed to hold for all generations of products.  If a future product results in the workload completing too quickly for a clean characterization the problem size will need to be increased.